### PR TITLE
Adding authentication via python code

### PR DESCRIPTION
**Pull request by lwasser**
hey there @dkhawk !! i've created a little python script that allows you to:

* authenticate with strava
* and then pulls down your summary activity data
if this all works the rest should be straight forward.

I have a secrets file that i'm not sure what to do with. open to any suggestions. `git crypt` or some other workflow to share it?
i made it with a dummy account so in theory it should be semi ok to share. 

the other issue is simply setting up the environment. i'm using to using a distribution called miniconda to manage environments and install tools. i wasn't sure what to setup in the readme for dev testing. open to any and all suggestions. miniconda is just really nice to envt management. i know others prefer a pip / virtual environment setup. 

happy coding! i'm learning so much. totally open to changing anything and everything. this is just a rough first stab. there is one manual authentication step which entails copying a code from the browser. i'm assuming we can grab this once users are directed to a site that we control!

### CODE REVIEW

File: process_strava/strava_clean.py [hunk 1]
```diff
@@ -0,0 +1,43 @@
+"""A module with helper functions to process data."""
+
+from shapely.geometry import LineString, Point
+import pandas as pd
+
+def swap(x):
```
Comment (lwasser) [inline process_strava/strava_clean.py:8]: this we can remove. it's for that weird - the data come down in lat, long which is y,x rather than x,y fix.

**Comment by lwasser** [comment]
@dkhawk whenever you get to this (no rush - i still have a lot of cleanup to do) i have some questions.

1. when i grab the data i can simplify the GPS data - there are three resolutions available via strava. i haven't tested the 3 options yet but will. i'll just need to know what works for you - i am happy to provide 3 different csv files to test with?

```

resolutionstring | The level of detail (sampling) in which this stream was returned May take one of the following values: low, medium, high
```

2. i sent an email with a few other questions. One was - should i swap the x,y values as they are actually y,x as returned!
More later!

**Comment by lwasser** [comment]
i just pushed another update. this one only grabs data from december on. turns out the new api with the free account has a 100 request limit vs 600. and i still havent figured out rate limiting so for now.. this returns both your activities and the runs with gps data from december. Next step will figure out how to rate limit ... a 15 minute wait is considerable so we may need the paid account and we may also want to consider how to pull in data just at first.

**Comment by dkhawk** [comment]
Great work.  Thanks, Leah!

**Pull request merged**